pykafka/producer.py

```diff
@@ -392,6 +393,7 @@ def _wait_all(self):
         log.info("Blocking until all messages are sent")
         while any(q.message_is_pending() for q in itervalues(self._owned_brokers)):
             time.sleep(.3)
+            self._cluster.handler.sleep(.3)
```

Comment:
Yup, that's an error. This branch is still in testing, and it doesn't work fully yet.